app/assets/javascripts/discourse/components/select-box.js.es6

```diff
@@ -0,0 +1,179 @@
+import { observes } from "ember-addons/ember-computed-decorators";
+
+export default Ember.Component.extend({
+  classNames: "select-box",
+
+  classNameBindings: ["expanded:is-expanded"],
+
+  attributeBindings: ['componentStyle:style'],
+  componentStyle: function() {
+    return Ember.String.htmlSafe(`width: ${this.get("maxWidth")}px`);
+  }.property("maxWidth"),
+
+  expanded: false,
+
+  caretUpIcon: "caret-up",
+  caretDownIcon: "caret-down",
+  icon: null,
+
+  data: [],
```

Comment:
If you assign a default this way, every instance of `select-box` will access the same array. You need to set the default in the `init()` method when it's an object. Same for `selectedData`, `filteredData` and any other array or object as a default.